src/client/java/net/neoforged/neoforge/client/ClientNeoForgeMod.java

```diff
@@ -157,7 +157,10 @@ static void onRegisterReloadListeners(AddClientReloadListenersEvent event) {
         event.addDependency(NeoForgeReloadListeners.BRANDING, VanillaClientListeners.FIRST);
 
         event.addListener(NeoForgeReloadListeners.OBJ_LOADER, ObjLoader.INSTANCE);
+
         event.addListener(NeoForgeReloadListeners.ENTITY_ANIMATIONS, AnimationLoader.INSTANCE);
+        // Animations need to be loaded before entity renderers are instantiated
+        event.addDependency(NeoForgeReloadListeners.ENTITY_ANIMATIONS, VanillaClientListeners.ENTITY_RENDERER);
```

Comment:
This should be fine given that models aren't actually baked until the entity renderer, but technically, the models themselves are loaded in the `ModelManager`, so it would be a bit safer to have this run before that.

Reply:
The `ModelManager` is irrelevant for entity models, it's only responsible for block and item models.

Reply:
They're loaded as part of the model manager since they are used in item special models.

Reply:
Right, I completely forgot that the item models have access to entity models. However, the item model baking happens entirely in the async prepare phase where the listener sorting has no effect (it only applies to the synchronous apply phase) and the animations wouldn't be accessible regardless, even if the animation loader's prepare phase is done early enough.
1.21.9 or whatever the next version will be is going to have a system in resource reloaders that may make it feasible to give item models access to animations but for now it's not feasible to do this.

Reply:
Hmm, we should probably make another issue to keep track of this then, since any model that uses the animation system cannot be used within an item model as the animations are baked in the mod constructor. Or, that animations should be stored in a lazy instance and that should be a docs issue.

Either makes sense to me until we revisit with the shared state in 21.9